### PR TITLE
Add api provider and userName to getAccount

### PR DESCRIPTION
**Pull request by danielailie**

### CODE REVIEW

File: erdjs/src/apiProvider.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  constructor(url: string, timeout?: number) {
+    this.url = url;
+    this.timeoutLimit = timeout || 1000;
+  }
+
+  /**
+   * Fetches the Network Stake.
+   */
+  async getNetworkStake(): Promise<NetworkStake> {
+    let response = await this.doGet(`stake`);
```
Comment (ccorcoveanu) [inline erdjs/src/apiProvider.ts:29]: Use regular quotes if we don't have any variables here

File: erdjs/src/apiProvider.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    let payload = response;
```
Comment (ccorcoveanu) [inline erdjs/src/apiProvider.ts:30]: This assignment is redundant, you can pass response directly to `fromHttpResponse `

File: erdjs/src/apiProvider.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    return NetworkStake.fromHttpResponse(payload);
+  }
+
+  private async doGet(resourceUrl: string): Promise<any> {
+    try {
+      let url = `${this.url}/${resourceUrl}`;
+      let response = await axios.get(url, { timeout: this.timeoutLimit });
+      let payload = response.data;
```
Comment (ccorcoveanu) [inline erdjs/src/apiProvider.ts:38]: You can just `return response.data`

File: erdjs/src/apiProvider.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+      return payload;
+    } catch (error) {
+      if (!error.response) {
```
Comment (ccorcoveanu) [inline erdjs/src/apiProvider.ts:42]: Maybe we can refactor this error handling into a function for more readability inside the `doGet`.

**Review by ccorcoveanu — CHANGES_REQUESTED** [review]

**Review by ccorcoveanu — APPROVED** [review]

**Review by ccorcoveanu — APPROVED** [review]

File: erdjs/src/networkStake.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    Object.assign(this, fresh);
+  }
+
+  /**
+   * Constructs a configuration object from a HTTP response (as returned by the provider).
+   */
+  static fromHttpResponse(payload: any): NetworkStake {
+    let networkStake = new NetworkStake();
+
+    networkStake.TotalValidators = Number(payload["totalValidators"]);
```
Comment (bogdan-rosianu) [inline erdjs/src/networkStake.ts:60]: shouldn't we have a check for the `payload`? maybe it does not contain all the keys

File: erdjs/src/interface.ts [hunk 6]
```diff
@@ -1,87 +1,102 @@
...
+  simulateTransaction(tx: Transaction): Promise<TransactionHash>;
 
-    /**
-     * Fetches the state of a {@link Transaction}.
-     */
-    getTransaction(txHash: TransactionHash): Promise<TransactionOnNetwork>;
+  /**
+   * Fetches the state of a {@link Transaction}.
+   */
+  getTransaction(txHash: TransactionHash): Promise<TransactionOnNetwork>;
```
[the remark this answers is not in the document]
Comment (danielailie) [reply erdjs/src/interface.ts:46]: I'll do this into another PR

File: erdjs/src/networkStake.ts (hunk 5 above)
Comment (danielailie) [reply erdjs/src/networkStake.ts:60]: if it doesn't contain it it returns null

**Review by miiu96 — APPROVED** [review]

**Pull request merged**